javascript/implementation/LogBackedStore.ts

```diff
@@ -62,10 +64,22 @@ export class LogBackedStore implements Store {
         }
         const unlockingFunction = await this.memoryLock.acquireLock();
         await this.pullDataFromFile();
+
+        this.fileWatcher =
+            watch(this.filename, async (eventType, filename) => {
+                await new Promise(r => setTimeout(r, 10));
+                const size = (await this.fileHandle.stat()).size;
+                if (eventType == "change" && size > this.redTo && !this.fileLocked) {
```

Comment:
You shouldn't just check to see if the file is locked by this process.  Instead, you need to acquire both the file lock (to ensure not data is written while you're reading) and the memory lock (to ensure that you're not pulling data while half way through an apply).  The memory lock that's part of the initialize function won't be active when your callback is called.

Reply:
You need to do something like is found in the addBundle function: 
```
        if (!this.exclusive)
            await this.lockFile(true);
```
and then unlock if not exclusive after pulling data.